### PR TITLE
Fix sending new data when codec runs out of input data

### PR DESCRIPTION
**Pull request by mlehtima**

### CODE REVIEW

File: tools/thumbnaild-video/libavthumbnailer.cpp [hunk 1]
```diff
@@ -204,6 +204,8 @@ QImage createVideoThumbnail(const QString &fileName, const QSize &requestedSize,
             } else if (ret != AVERROR(EAGAIN)) {
                 av_packet_unref(&packet);
                 return image;
+            } else {
+                break;
```
Comment (Thaodan) [inline tools/thumbnaild-video/libavthumbnailer.cpp:209]: Should there be some warning or error handling from the codec?

The fix works good on Xperia 1.

Comment (mlehtima) [reply tools/thumbnaild-video/libavthumbnailer.cpp:209]: Added a comment to clarify the code.

**Review by Thaodan — APPROVED** [review]

File: tools/thumbnaild-video/libavthumbnailer.cpp [hunk 2]
```diff
@@ -204,6 +204,9 @@ QImage createVideoThumbnail(const QString &fileName, const QSize &requestedSize,
             } else if (ret != AVERROR(EAGAIN)) {
                 av_packet_unref(&packet);
                 return image;
+            } else {
+                // When EAGAIN is returned codec needs more data
+                break;
```
Comment (monich) [inline tools/thumbnaild-video/libavthumbnailer.cpp:209]: Is it ok that `av_packet_unref(&packet)` isn't being done in this case? And is `thumbnailer.frame` guaranteed to be pointing to a valid frame?

Comment (mlehtima) [reply tools/thumbnaild-video/libavthumbnailer.cpp:209]: This break only causes the inner loop to break and the av_packet_unref(&packet) is called a few lines later at end of the outer loop. In the case where the inner loop breaks here thumbnailer.frame is not set yet as next frame will be attempted during next iteration of the outer loop.

Comment (monich) [reply tools/thumbnaild-video/libavthumbnailer.cpp:209]: ok, sounds sane!

**Pull request merged**